utils/users/filterAllowedUserFields.js

```diff
@@ -0,0 +1,35 @@
+const { allowedUserFieldsForUpdate } = require('~/consts/user')
+
+const isObjectEmpty = (obj) => Object.keys(obj).length === 0
+
+const isAllowedField = (key, allowedFields = {}) => {
+  if (isObjectEmpty(allowedFields)) return key in allowedUserFieldsForUpdate
+  return key in allowedFields
+}
+const isFieldObject = (key) => typeof key == 'object'
+
+const operateNestedField = ({ key, fields, updatedFields }) => {
+  const nestedFields = fields[key]
+  for (const nestedKey in nestedFields) {
+    if (!updatedFields[key]) updatedFields[key] = {}
+    if (!isAllowedField(nestedKey, allowedUserFieldsForUpdate[key])) continue
+
+    updatedFields[key][nestedKey] = nestedFields[nestedKey]
+  }
+}
+
+const operateField = ({ key, fields, updatedFields }) => {
+  if (isAllowedField(key)) {
+    if (isFieldObject(fields[key])) operateNestedField({ key, fields, updatedFields })
+    else updatedFields[key] = fields[key]
+  }
+}
+const filterAllowedUserFields = (fields) => {
```

Comment:
You can simplify all this code by replacing allowedFieldsForUpdate object with array of allowed fields, for example ['firstName', 'lastName']. Then, instead of all this methods you can just use "for(key in fields)" and check if allowedFieldsForUpdate array includes the key, and if so, push it to new object, in your case to updatedFields

Reply:
And for the future, use {} with if

Reply:
deleted nested fields @Roman-Peretiatko